service/raft/raft_group0.cc

```diff
@@ -129,17 +138,14 @@ raft_server_for_group raft_group0::create_server_for_group(raft::group_id gid,
 }
 
 future<group0_info>
-raft_group0::discover_group0(raft::server_address my_addr) {
-    std::vector<raft::server_address> seeds;
-    seeds.reserve(_gossiper.get_seeds().size());
-    for (auto& seed : _gossiper.get_seeds()) {
-        if (seed == _gossiper.get_broadcast_address()) {
-            continue;
-        }
-        seeds.push_back({.info = inet_addr_to_raft_addr(seed)});
-    }
-
-    _group0.emplace<persistent_discovery>(co_await persistent_discovery::make(my_addr, std::move(seeds), _qp));
+raft_group0::discover_group0(raft::server_id my_id, raft::server_info my_info, const std::vector<raft::server_info>& seeds) {
+    // Note: raft::server_address contains a .can_vote flag which is irrelevant for discovery
+    raft::server_address my_addr{.id = my_id, .info = std::move(my_info)};
+    std::vector<raft::server_address> seed_addrs;
+    std::transform(seeds.begin(), seeds.end(), std::back_inserter(seed_addrs),
+            [] (const raft::server_info& info) { return raft::server_address {.info = info}; });
```

Comment:
I personally dislike std::transform and std::back_inserter. Making such changes in the same commit which moves the code around and renames methods also makes it difficult to see what really is going on in the commit.